### DISCUSSION TITLE
UnboundLocalError: local variable 'out_file' referenced before assignment

### DISCUSSION THREAD

**Discussion by arunrajan6600** [d8ahazard/sd_dreambooth_extension#866] [Q&A]
Hi,
newbie here. When i try to train using custom set of images in SD with dreambooth with LORA on, ticked 
'Generate a .ckpt file when saving during training" in savings settings, the ckpt file is not saving and giving me the below error but the training continues 

Num batches each epoch = 132
  Num Epochs = 150
  Batch Size Per Device = 1
  Gradient Accumulation steps = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Text Encoder Epochs: 112
  Total optimization steps = 9900
  Total training steps = 19800
  Resuming from checkpoint: False
  First resume epoch: 0
  First resume step: 0
  Lora: True, Adam: True, Prec: bf16
  Gradient Checkpointing: True
  EMA: False
  UNET: True
  Freeze CLIP Normalization Layers: False
  LR: 0.0002
  LoRA Text Encoder LR: 0.0002
  V2: False
Compiling Checkpoint:  25%|███▎         | 1/4 [00:00<?, ?it/s, loss=0.00232, loss_avg=0.102, lr=0.0002, vram_usage=4.7]Exception saving checkpoint/model: local variable 'out_file' referenced before assignment
Traceback (most recent call last):
  File "D:\Stable Diffusion\stable-diffusion-webui-master\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 690, in save_weights
    compile_checkpoint(args.model_name, reload_models=False, lora_path=out_file, log=False,
UnboundLocalError: local variable 'out_file' referenced before assignment

Any help to solve this issue?

**Comment by nicolasjavierp** [comment]
You are missing a path, doublecheck the Sample/DataSet/TrainingSet/Lora Directory

**Reply by Vektor369** [reply]
Where are those paths supposed to be?

**Comment by Vektor369** [comment]
Yeah, I'm getting the same thing.

Compiling Checkpoint:  25%|██▊        | 1/4 [00:00<?, ?it/s, loss=0.0451, loss_avg=0.0863, lr=0.000156, vram_usage=4.9]Exception saving checkpoint/model: local variable 'out_file' referenced before assignment
Traceback (most recent call last):
  File "F:\Odyssey\AI\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 690, in save_weights
    compile_checkpoint(args.model_name, reload_models=False, lora_path=out_file, log=False,
UnboundLocalError: local variable 'out_file' referenced before assignment

It goes through Generating Samples, does some Steps, then errors, and continues. Only the loss, loss_avg, and lr are different each time.
The training completes though, so is this anything I need to worry about? I can't find any evidence of any of the directories you mentioned. Were they supposed to be created upon installation? Where are they supposed to be?
